src/daemon/rpmostreed-transaction-livefs.c

```diff
@@ -628,29 +696,61 @@ livefs_transaction_execute_inner (LiveFsTransaction *self,
   if (!checkout_add_usr (repo, deployment_dfd, diff, target_csum, cancellable, error))
     return FALSE;
 
-  /* Start replacing the rpmdb. First, ensure the temporary dir for the new
-     version doesn't exist */
-  if (!glnx_shutil_rm_rf_at (deployment_dfd, orig_rpmdb_path, cancellable, error))
-    return FALSE;
-  /* Check out the new rpmdb */
-  { OstreeRepoCheckoutAtOptions rpmdb_checkout_opts = { .mode = OSTREE_REPO_CHECKOUT_MODE_NONE,
-                                                        .no_copy_fallback = TRUE,
-                                                        .subpath = "/usr/share/rpm" };
-
-
-    if (!ostree_repo_checkout_at (repo, &rpmdb_checkout_opts, deployment_dfd, orig_rpmdb_path,
-                                  target_csum, cancellable, error))
+  /* And files that we always need to replace; the rpmdb, /usr/lib/passwd. We
+   * make a tmpdir just for this since it's a more convenient place to put
+   * temporary files/dirs without generating tempnames.
+   */
+  { g_auto(GLnxTmpDir) replace_tmpdir = { 0, };
+    if (!glnx_mkdtempat (ostree_repo_get_dfd (repo), "tmp/rpmostree-livefs.XXXXXX", 0700,
+                         &replace_tmpdir, error))
       return FALSE;
+
+    const char *replace_paths[] = { "/usr/share/rpm", "/usr/lib/passwd", "/usr/lib/group" };
+    for (guint i = 0; i < G_N_ELEMENTS(replace_paths); i++)
+      {
+        const char *replace_path = replace_paths[i];
+        if (!replace_subpath (repo, deployment_dfd, &replace_tmpdir,
+                              target_csum, replace_path,
+                              cancellable, error))
+          return FALSE;
+      }
   }
-  /* Now, RENAME_EXCHANGE the two */
-  if (glnx_renameat2_exchange (deployment_dfd, "usr/share/rpm", deployment_dfd, orig_rpmdb_path) < 0)
-    return glnx_throw_errno_prefix (error, "%s", "rename(..., RENAME_EXCHANGE) for rpmdb");
-  /* And nuke the old one */
-  if (!glnx_shutil_rm_rf_at (deployment_dfd, orig_rpmdb_path, cancellable, error))
-    return FALSE;
 
   rpmostree_output_task_end ("done");
 
+  rpmostree_output_task_begin ("Running systemd-tmpfiles for /run,/var");
```

Comment:
We can easily make this conditional, right? We're already analyzing file paths before we being, so maybe let's look for `tmpfiles.d` files too?